### PR TITLE
feat(button-group): update interface

### PR DESCRIPTION
**Pull request by sungik-choi**
## Self Checklist

- [x] I wrote a PR title in **English** and added an appropriate **label** to the PR.
- [x] I wrote the commit message in **English** and to follow [**the Conventional Commits specification**](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I [added the **changeset**](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) about the changes that needed to be released. (or didn't have to)
- [x] I wrote or updated **documentation** related to the changes. (or didn't have to)
- [x] I wrote or updated **tests** related to the changes. (or didn't have to)
- [x] I tested the changes in various browsers. (or didn't have to)
  - Windows: Chrome, Edge, (Optional) Firefox
  - macOS: Chrome, Edge, Safari, (Optional) Firefox

## Related Issue

- #1733 

## Summary

Update `ButtonGroup` interface

## Details

- margin prop을 지원하도록 합니다.
- named export, jsdoc 추가 등

### Breaking change? (Yes/No)

No

### CODE REVIEW

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.types.ts [hunk 1]
```diff
@@ -5,16 +5,16 @@ import {
...
   withoutSpacing?: boolean
 }
 
-export default interface ButtonGroupProps extends
+export interface ButtonGroupProps extends
   AlphaBezierComponentProps,
   ChildrenProps,
-  Pick<StackProps, 'justify'>,
-  ButtonGroupOptions {}
+  Omit<StackProps, 'as' | 'role' | 'direction' | 'spacing'>,
```
Comment (sungik-choi) [inline packages/bezier-react/src/components/ButtonGroup/ButtonGroup.types.ts:19]: LayoutProps 지원은 하지 말 것

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -2,27 +2,23 @@ import React, { forwardRef } from 'react'
...
+export const ButtonGroup = forwardRef<HTMLDivElement, ButtonGroupProps>(function ButtonGroup({
+  children,
+  justify = 'center',
+  withoutSpacing = false,
+  ...props
+}, forwardedRef) {
   return (
     <Stack
+      {...props}
+      as="div"
```
Comment (yangwooseong) [inline packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx:25]: div가 기본값이라 안써도 되지 않나요?

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -4,6 +4,15 @@ import { Stack } from '~/src/components/Stack'
 
 import type { ButtonGroupProps } from './ButtonGroup.types'
 
+/**
+ * `ButtonGroup` is a component that groups buttons together.
+ *
+ * @example
```
Comment (yangwooseong) [inline packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx:10]: \```tsx 를 안써도 VSCode가 code block 으로 인지를 해주네요

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx (hunk 2 above)
Comment (sungik-choi) [reply packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx:25]: 타입 무시하면 Stack prop을 오버라이드할 수 있어서 작성했는데, 생각해보니 다른 속성도 마찬가지라서 빼도 괜찮겠네요.

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx [hunk 4]
```diff
@@ -8,10 +8,12 @@ import type { ButtonGroupProps } from './ButtonGroup.types'
  * `ButtonGroup` is a component that groups buttons together.
  *
  * @example
+ * ```
```
Comment (yangwooseong) [inline packages/bezier-react/src/components/ButtonGroup/ButtonGroup.tsx:11]:
```suggestion
 * ```tsx
```

**Review by yangwooseong — APPROVED** [review]

**Pull request merged**